### DISCUSSION TITLE
Multiphase Porous Flow supercritical water

### DISCUSSION THREAD

**Discussion by RotmanC** [idaholab/moose#28055] [Q&A Modules: Porous Flow]
Hello.

Following issues about Multiphase Porous Flow (pliquid/gas) [27159](https://github.com/idaholab/moose/discussions/27159#discussioncomment-9205632) and [27457](https://github.com/idaholab/moose/issues/27457), I found it doesn't work for the evolution of 1 phase of water to 3 phases (liquid, gas and supercritical conditions).

The example has liquid as initial condition (from times -3.15576e+10 to 0)  and then start heating the bottom of the model. It should modify the saturation of liquid and gas until arrives at supercritical conditions of water (pore pressure > 22 MPa, temperature > 374 ºC). The result is 3 phases: liquid, gas and supercritical conditions.

When I use water97 properties (`Water97FluidProperties`) it shows this error:
```
Time Step 0, time = -3.15576e+10

Postprocessor Values:
+----------------+----------------+----------------+----------------+----------------+
| time           | density_gas    | density_water  | enthalpy       | enthalpy_gas   |
+----------------+----------------+----------------+----------------+----------------+
|  -3.155760e+10 |   0.000000e+00 |   1.011352e+03 |   1.439250e+05 |   0.000000e+00 |
+----------------+----------------+----------------+----------------+----------------+
+----------------+----------------+----------------+----------------+----------------+
| time           | enthalpy_water | liquid_mass    | pgas           | pwater         |
+----------------+----------------+----------------+----------------+----------------+
|  -3.155760e+10 |   1.439250e+05 |   1.061916e+10 |   0.000000e+00 |   3.463821e+07 |
+----------------+----------------+----------------+----------------+----------------+
+----------------+----------------+----------------+----------------+----------------+
| time           | sg             | sw             | temperature    | vapor_mass     |
+----------------+----------------+----------------+----------------+----------------+
|  -3.155760e+10 |   0.000000e+00 |   1.000000e+00 |   2.999896e+02 |   0.000000e+00 |
+----------------+----------------+----------------+----------------+----------------+
+----------------+----------------+----------------+
| time           | viscosity_gas  | viscosity_water|
+----------------+----------------+----------------+
|  -3.155760e+10 |   1.000000e+00 |   8.531495e-04 |
+----------------+----------------+----------------+


Time Step 1, time = -3.1526e+10, dt = 3.15576e+07
    |residual|_2 of individual variables:
                    pliquid: 0
                    h:       4.17257
 0 Nonlinear |R| = 4.172568e+00
      0 Linear |R| = 4.172568e+00
      1 Linear |R| = 8.819981e-16

Enthalpy 1.33841e+07 is out of range in water97: inRegionPH()
To recover, the solution will fail and then be re-attempted with a reduced time step.

Nonlinear solve did not converge due to DIVERGED_FNORM_NAN iterations 0
 Solve Did NOT Converge!
```

When I use tabulated properties to read a csv file of water properties obtained with [Coolprop](http://www.coolprop.org/), it shows this error:
```
*** ERROR ***
/home/geoen/projects/rotman/equi/Equi_8_2D_TH_vapor.i:259.3:
The following error occurred in the FluidProperties 'water_tab' of type TabulatedBicubicFluidProperties.

virtual libMesh::Real SinglePhaseFluidProperties::triplePointPressure() const not implemented.
```

You can see the example here: [https://github.com/RotmanC/SC_evolution/blob/main/Equi_8_2D_TH_vapor.i](https://github.com/RotmanC/SC_evolution/blob/main/Equi_8_2D_TH_vapor.i)

How I can consider this evolution of fluid phases of water?

Thank you for your attention!

Rotman

**Comment by GiudGiud** [comment]
Hello

If you want to use the TabulatedFluidProperties, you'll need to add a little logic in there for the triplePointPressure because we currently do not read a tabulation for it. 

For IAPWS97 it seems you are hitting an out of range region
```
Enthalpy 1.33841e+07 is out of range in water97: inRegionPH()
```
is that physical? what is the pressure and temperature for this?
What can happen is that the nonlinear solver takes the solution way out of the physical range on its way to convergence.
Smaller time steps, using Bounds, and more relaxation can alleviate that

Guillaume

**Reply by RotmanC** [reply]
Dear @GiudGiud .
I didn't reply before because I was trying to fix it using smaller time steps, bounds and more soft changes, but it doesn't make any variation of temperature shapes. I would have to create convection cells around once the bottom of the model starts heating and it should create 2 phases (liquid and gas) areas too.

This environment has more than 374ºC of temperature and 22.1 MPa for pressure.

If you need some help trying to fix it please let me know.

**Reply by GiudGiud** [reply]
Hello

We have a fix in the works for using triplePointPressure in TFP
It's in this PR which is currently queued up
https://github.com/idaholab/moose/pull/27506
Please feel free to try this
